### ISSUE TITLE
Rule Change: sort-keys should include the option of compatibility with `localeCompare` order

### ISSUE THREAD

**Issue by bitjson** [eslint/eslint#18220]
### What rule do you want to change?

sort-keys

### What change do you want to make?

Generate more warnings

### How do you think the change should be implemented?

A new option

### Example code

```js
const obj = {
  errors: ['Some parsing error.']
  errorType: 'parse',
};
```

### What does the rule currently do for this code?

Currently, `sort-keys` will warn: `Expected object keys to be in ascending order. 'errorType' should be before 'errors'.`.

While  [`localeCompare`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/localeCompare) sorts these keys in the reverse order: `Object.keys(obj).sort((a, b) => a.localeCompare(b, 'en'));` produces `'errors', 'errorType'`.

### What will the rule do after it's changed?

The `sort-keys` rule could offer the option to sort keys with each capital letter following its lowercase equivalent (the default settings of `localeCompare`), so the example code would not fail to lint.

One backwards-compatible option might be to add a `caseFirst` option (like [that of Intl.Collator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/Collator/Collator#casefirst)) which if set to `lower` would replicate the default settings of `localeCompare`, allow the option of `upper`, and default to `false` to avoid changing the current behavior.

### Participation

- [ ] I am willing to submit a pull request to implement this change.

### Additional comments

Some discussion about a similar proposal for a typescript-eslint An discussing covering the same problem with another rule: https://github.com/typescript-eslint/typescript-eslint/issues/5989
